lib/ecto/repo.ex

```diff
@@ -1537,6 +1537,28 @@ defmodule Ecto.Repo do
   inserting a struct with associations and using the `:on_conflict` option
   at the same time is not recommended, as Ecto will be unable to actually
   track the proper status of the association.
+
+  ## Advanced Upserts
+
+  Using an `Ecto.Query` for `on_conflict` can allow us to use more advanced
+  database features. For example, PostgreSQL supports conditional upserts like
+  `DO UPDATE SET title = EXCLUDED.title WHERE EXCLUDED.version > post.version`.
+  This means that the title will be updated only if the proposed row has a
+  greater version value than the existing row.
+
+  Ecto can support this as follows:
+
+      conflict_query =
+        from(p in Post,
+          update: [set: [title: fragment("EXCLUDED.title")]],
```

Comment:
not sure if it's too pedantic for an example, but probably the version should be updated as well?

Reply:
@greg-rychlewski Good point. I expanded the example to update the version. I also made the title less boring. 😄